### PR TITLE
[stable/redis] Add support to execute Redis as root

### LINKED ISSUE #15246: [stable/redis] error: failed switching to "redis": unable to find user redis: no matching entries in passwd file
**Describe the bug**
Master and Slave pods keep crashing due to the below error.

**Version of Helm and Kubernetes**: Kubernetes: v1.14, Helm v2.9.1

**Which chart**: stable/redis

**What happened**:
```
Mohammeds-MacBook-Pro:further-exchange mfadin$ kubectl logs gogs-redis-master-1
INFO  ==> ** Starting Redis **
error: failed switching to "redis": unable to find user redis: no matching entries in passwd file

Mohammeds-MacBook-Pro:further-exchange mfadin$ kubectl logs gogs-redis-slave-0
INFO  ==> ** Starting Redis **
error: failed switching to "redis": unable to find user redis: no matching entries in passwd file
```

**What you expected to happen**: Pods to run

**How to reproduce it** (as minimally and precisely as possible):
```
helm upgrade  --tls gogs-redis --set master.persistence.size=20Gi --set master.persistence.storageClass=ibmc-file-bronze --set securityContext.enabled=true --set securityContext.fsGroup=0 --set securityContext.runAsUser=0  --set volumePermissions.enabled=true  stable/redis
```
**Anything else we need to know**: None

### PR DESCRIPTION
**Pull request by miguelaeh**
#### What this PR does / why we need it:
Change the command to start redis so when deploying as root it creates the redis user.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #15246

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by tompizmor** [comment]
@macabrera it seems you didn't set the proper version

**Comment by miguelaeh** [comment]
Hi @tompizmor ,
You are right, I forgot to update my local upstreamed. Now it should be the correct version.

**Review by tompizmor — CHANGES_REQUESTED** [review]
I think that starting the container as root or not should be a feature in the docket image itself and not additional logic in the helm chart.

Can you try to start redis as root just using docker or docker-compose and see if it works? If it doesn't, we should fix it in the image and then try again with the current chart.

As we are overriding the entrypoint in the chart, it is possible that we also need your changes, but I want to make sure we don't need to update the image too.

**Comment by miguelaeh** [comment]
Hi @tompizmor ,
I already checked it before the changes and using docker-compose it start correctly. I think the problem is we are overriding the command, in any case the logic to do that on the image is on the `setup.sh` file that is not used on the `Dockerfile`.

**Comment by tompizmor** [comment]
@miguelaeh 

>  in any case the logic to do that on the image is on the setup.sh file that is not used on the Dockerfile.

You mean that the `setup.sh` is only called from the `entrypoint.sh` and we are not calling `entrypoint.sh` from this chart right? But the `setup.sh` is included in the image and used by default right?

File: stable/redis/templates/redis-slave-statefulset.yaml [hunk 1]
```diff
@@ -76,6 +76,10 @@ spec:
         - /bin/bash
         - -c
         - |
+          {{- if (eq .Values.securityContext.runAsUser 0.0) }}
+          useradd redis
+          chown -R redis /data
```
Comment (tompizmor) [inline stable/redis/templates/redis-slave-statefulset.yaml]: use `{{ .Values.persistence.path }}` instad of `/data`

Comment (miguelaeh) [reply stable/redis/templates/redis-slave-statefulset.yaml]: Applied.

**Pull request merged**